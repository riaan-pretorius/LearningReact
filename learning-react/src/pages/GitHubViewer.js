import React, { Component } from "react";
import UserSearch from "../components/UserSearch";

class Home extends Component {
  render() {
    return <UserSearch />;
  }
}

export default Home;
