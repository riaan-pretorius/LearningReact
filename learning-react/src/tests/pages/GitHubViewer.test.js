import React from "react";
import GitHubViewer from "../../pages/GitHubViewer";
import { mount, shallow } from "enzyme";
import renderer from "react-test-renderer";

test("GitHubViewer Test Snapshot", () => {
  const component = renderer.create(<GitHubViewer />);
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

test("GitHubViewer Test", () => {
  const page = mount(<GitHubViewer />);

  expect(page.find(".userSearch").exists()).toBeTruthy();
  expect(page.find("input").exists()).toBeTruthy();
  expect(page.find("button").exists()).toBeTruthy();

  page.find("input").text("riaan-pretorius");
  page.find("button").simulate("click");

  // setTimeout(() => {
  //   expect(page.find(".userResults").exists()).toBeTruthy();
  //   expect(page.find("li").exists()).toBeTruthy();
  //   page.find("li").simulate("click");
  //   setTimeout(() => {
  //     expect(page.find(".userProfile").exists()).toBeTruthy();
  //   });
  // }, 2000);
});
