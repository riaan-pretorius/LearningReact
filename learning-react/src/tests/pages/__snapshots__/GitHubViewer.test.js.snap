// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubViewer Test Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col col-sm-4"
    style={
      Object {
        "borderRight": "5px",
        "borderRightStyle": "groove",
      }
    }
  >
    <form
      className=""
    >
      <div
        className="userSearch input-group"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Search github user"
          type="text"
        />
        <div
          className="input-group-btn"
        >
          <button
            className="btn btn-secondary disabled"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </form>
    
  </div>
  <div
    className="col col-sm-8"
  >
    
  </div>
</div>
`;
